src/dbnode/storage/shard.go

```diff
@@ -1804,6 +1804,22 @@ func (s *dbShard) FetchBlocksMetadataV2(
 	return result, nil, nil
 }
 
+func (s *dbShard) PrepareBootstrap() error {
+	s.flushState.RLock()
+	flushStateBootstrapped := s.flushState.bootstrapped
+	s.flushState.RUnlock()
+	if flushStateBootstrapped {
+		return nil // Already prepared.
+	}
+
+	// Iterate flushed time ranges to determine which blocks are retrievable.
+	// NB(r): This must be done before bootstrap since during bootstrapping
```

Comment:
Is it worth adding some sort of flag that will ensure this is the case, or maybe setting `s.flushState` to some sort of `bootstrappingPrepared` state?

Reply:
Yup I've changed `bootstrapped` bool to `initialized` bool to track this.